### PR TITLE
Add Component for Switching Date Display in Entity Cards; closes #2598

### LINKED ISSUE #2598: Add Component for Switching Date Display in Entity Cards
Currently the `DateCard` and `TicketCard` components are using the `CalendarDateRange` component for their sidebar content but this does not change views when the "**display**" filter option is changed.

- create a new component in `assets\src\application\ui\calendars\dateDisplay` called `CalendarDateSwitcher` (open to better ideas for name)

- implement logic similar to that in `assets\ZZZ\editor\events\dates-and-times\editor-date\grid-view\event-date-calendar-date.js` but abstract things so that it will work with any entity simply by providing the necessary props such as:

     - `showDate` (or `displayDate`)

     - `startDate`

     - `endDate`

     - `headerText` & `footerText`

     - `sidebarColorClass`

- utilize this new component in both `DateCard` and `TicketCard`

- wire up to the filter bar "**display**" filter and ensure that dates switch and display correctly as the filter changes

### PR DESCRIPTION
**Pull request by nicolad**
## Problem this Pull Request solves
- Move old functionality and adapted in `CalendarDateSwitcher`
- Closes https://github.com/eventespresso/event-espresso-core/issues/2598

## How has this been tested
Peek 2020-03-25 18-19

### CODE REVIEW

File: assets/src/application/ui/calendars/dateDisplay/CalendarDateSwitcher/index.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	headerPrefix?: string;
+	headerText?: string;
+	showDate?: boolean;
+	startDate: string;
+	statusClassName?: string;
+}
+
+const CalendarDateSwitcher: React.FC<Props> = React.memo(
+	({ displayDate, footerText, headerPrefix, headerText, statusClassName, ...props }) => {
+		const className = classNames('ee-editor-date-calendar-sidebar', statusClassName);
```
Comment (tn3rb) [inline assets/src/application/ui/calendars/dateDisplay/CalendarDateSwitcher/index.tsx]: I may have made this mistake in the other calendar components in this folder,
but anything in the application folder should not reference a specific use case like "editor-date"
as well, if the idea is for this to be reusable anywhere, then it also should not include "sidebar" in the class name

what if we just pass `statusClassName` through to the child components?
this doesn't add any markup anyways

File: assets/src/application/ui/calendars/dateDisplay/CalendarDateSwitcher/style.scss [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+			align-items: center;
+			flex-flow: row nowrap;
+			justify-content: center;
+
+			&__divider {
+				padding: 0 var(--ee-padding-big);
+			}
+		}
+	}
+}
```
Comment (tn3rb) [inline assets/src/application/ui/calendars/dateDisplay/CalendarDateSwitcher/style.scss]: why is the css from CalendarDateRange copied into this component?!!?!
it's also not even loaded?!?

hmmm... did you copy the CalendarDateRange folder and then forget to delete this? 🤔

Comment (nicolad) [reply assets/src/application/ui/calendars/dateDisplay/CalendarDateSwitcher/style.scss]: yup, just removed

File: assets/src/application/ui/calendars/dateDisplay/CalendarDateSwitcher/index.tsx (hunk 1 above)
Comment (nicolad) [reply assets/src/application/ui/calendars/dateDisplay/CalendarDateSwitcher/index.tsx]: Thx, have cleaned up that.

File: assets/src/application/ui/calendars/dateDisplay/CalendarDateSwitcher/index.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+		switch (displayDate) {
+			case 'end':
+				return (
+					<BiggieCalendarDate
+						className={className}
+						date={endDate}
+						headerText={`${headerPrefix || ''}${headerText || __('ends')}`}
+						footerText={footerText}
+					/>
```
Comment (tn3rb) [inline assets/src/application/ui/calendars/dateDisplay/CalendarDateSwitcher/index.tsx]: can we add the `showTime` prop to all of the `BiggieCalendarDate` components plz?

```suggestion
					<BiggieCalendarDate
						className={className}
						date={endDate}
						headerText={`${headerPrefix || ''}${headerText || __('ends')}`}
						footerText={footerText}
						showTime
					/>
```

File: assets/src/application/ui/calendars/dateDisplay/CalendarDateSwitcher/index.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	startDate: string;
+}
+
+const CalendarDateSwitcher: React.FC<Props> = React.memo(
+	({
+		className,
+		displayDate = DisplayStartOrEndDate.start,
+		footerText,
+		headerPrefix,
+		headerText,
```
Comment (tn3rb) [inline assets/src/application/ui/calendars/dateDisplay/CalendarDateSwitcher/index.tsx]: let's just use the `headerText` prop only 

then later on we'll replace the text being fed into after working on https://github.com/eventespresso/event-espresso-core/projects/6#card-35163011

**Comment by nicolad** [comment]
@tn3rb for some reasons after rebase this PR was closed;
have opened a new one: https://github.com/eventespresso/event-espresso-core/pull/2614

**Pull request closed without merging**